kimchi/src/circuits/lookup/lookups.rs

```diff
@@ -263,13 +261,30 @@ impl<F: Copy> JointLookup<SingleLookup<F>, LookupTableID> {
     }
 }
 
+#[derive(Clone, Serialize, Deserialize)]
 pub enum LookupPattern {
     ChaCha,
     ChaChaFinal,
     LookupGate,
 }
 
 impl LookupPattern {
+    pub fn max_lookups_per_row(&self) -> usize {
```

Comment:
missing doc, suggesting: "Returns the maximum number of lookups used by this pattern on a single row"